### ISSUE TITLE
How to get metrics (i.e. RMSE) on prediction set when running TinyTimeMixer

### ISSUE THREAD

**Issue by matsuobasho** [ibm-granite/granite-tsfm#50]
I see loss metrics when I run `zeroshot_trainer.evaluate(test_dataset)`.

I also see the predictions when I run 
`predictions = zeroshot_trainer.predict(test_dataset)`

However, the predictions inside that object are in an odd format that I can't quite figure out.
i.e. `len(predictions[0])`
returns 5.  No idea what the 5 elements of that tuple are.  

What is the best way to get a metric on the test set (like RMSE)?
A corollary is to update the prediction documentation to explain the contents of the predict output (but happy to create a separate issue for this).

**Comment by wgifford** [comment]
For the contents of `predictions` take a look at: https://github.com/ibm-granite/granite-tsfm/issues/29

**Comment by matsuobasho** [comment]
Thanks, that helps!  So it sound like manually calculating the RMSE from this prediction object is the way to go, if I want metrics on my test set?

**Comment by wgifford** [comment]
For metrics you can define a callable to be used in the `trainer.evaluate` call: https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Trainer

I am not sure I have an example to share at the moment. If you could try it and share your result that would be helpful.

**Comment by wgifford** [comment]
Hi @matsuobasho I took a look in to this -- here is what seems to be working for me:

```
import numpy as np
from transformers import EvalPrediction, Trainer, TrainingArguments

from tsfm_public import TinyTimeMixerForPrediction


model = TinyTimeMixerForPrediction.from_pretrained("ibm-granite/granite-timeseries-ttm-v1").to("cpu")

# custom RMSE function
def compute_metrics(p: EvalPrediction):
    return {"rmse": np.sqrt(np.square(p.predictions[0] - p.label_ids).mean())}

t = Trainer(
    model,
    args=TrainingArguments(output_dir="tmp", use_cpu=True, label_names=["future_values"]),
    compute_metrics=compute_metrics,
)

t.evaluate(dset_test)
```
Example output:
```
{'eval_loss': 0.42966070771217346,
 'eval_rmse': 0.6554850339889526,
 'eval_runtime': 12.7928,
 'eval_samples_per_second': 264.915,
 'eval_steps_per_second': 33.144}
```
